src/edg.rs

```diff
@@ -372,28 +490,60 @@ impl<B: Broker<V> + Debug, G: ExtendedDependencyGraph<V> + Send + Sync + Debug,
             None => {
                 // UNEXPLORED
                 // Line 6
+                trace!(
+                    ?edge,
+                    ?weight,
+                    assignment = "UNEXPLORED",
+                    "processing negation edge"
+                );
                 self.add_depend(&edge.target, Edges::NEGATION(edge.clone()));
-                let (weight1, weight2) = weight
-                    .split()
-                    .unwrap_or_else(|_| panic!("Worker {} ran out of weight", self.id));
-                self.broker
-                    .send(self.id, Message::NEGATION(edge.clone(), weight1));
-                self.explore(&edge.target, weight2);
+                self.queue_negation(edge.clone(), weight.clone());
+                self.explore(&edge.target, weight.clone());
```

Comment:
the same weight is used twice. It should be `.split()` to avoid introducing extra into the system